### PR TITLE
[Analytics] Code refactor & covered unit test in Analytics/ReportXml/QueryFactory

### PR DESCRIPTION
**Pull request by srsathish92**
### Description (*)
This PR utilise the existing `JSON class` instead of  PHP `json_encode` , `json_decode`, reused the Query Object create method & covered with unit test..

### Manual testing scenarios (*)
`vendor/bin/phpunit -c dev/tests/unit/phpunit.xml.dist app/code/Magento/Analytics/Test/Unit/ReportXml/QueryFactoryTest.php`

### Questions or comments
If any please let me know your feedback's

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Analytics/ReportXml/QueryFactory.php [hunk 1]
```diff
@@ -101,12 +113,29 @@ private function constructQuery($queryName)
...
+
+    /**
+     * Create query class using objectmanger
+     *
+     * @param  Select $select
+     * @param  string $connection
+     * @param  array  $queryConfig
+     * @return Query
+     */
+    private function createQueryObject($select, $connection, $queryConfig)
```
Comment (ihor-sviziev) [inline app/code/Magento/Analytics/ReportXml/QueryFactory.php]: Could you define argument types, return type and move this new method below public methods ?

File: app/code/Magento/Analytics/ReportXml/QueryFactory.php [hunk 2]
```diff
@@ -45,36 +49,44 @@ class QueryFactory
...
+        array $assemblers,
+        Json $jsonSerializer = null
     ) {
         $this->config = $config;
         $this->selectBuilderFactory = $selectBuilderFactory;
         $this->assemblers = $assemblers;
         $this->queryCache = $queryCache;
         $this->objectManager = $objectManager;
         $this->selectHydrator = $selectHydrator;
+        $this->jsonSerializer = $jsonSerializer ?: ObjectManager::getInstance()->get(Json::class);
```
Comment (ihor-sviziev) [inline app/code/Magento/Analytics/ReportXml/QueryFactory.php]: Please remove fallback to object manager and make this parameter not optional. There it’s not needed as this class in not marked as `@api`

File: app/code/Magento/Analytics/ReportXml/QueryFactory.php [hunk 3]
```diff
@@ -45,36 +49,44 @@ class QueryFactory
...
+     * @var Json|null
+     */
+    private $jsonSerializer;
+
     /**
      * QueryFactory constructor.
      *
-     * @param CacheInterface $queryCache
-     * @param SelectHydrator $selectHydrator
+     * @param CacheInterface         $queryCache
```
Comment (ihor-sviziev) [inline app/code/Magento/Analytics/ReportXml/QueryFactory.php]: Please revert formatting change for all arguments, like this
```suggestion
     * @param CacheInterface $queryCache
```
Please do it for all changes methods

File: app/code/Magento/Analytics/Test/Unit/ReportXml/QueryFactoryTest.php [hunk 4]
```diff
@@ -3,161 +3,209 @@
...
             ]
         );
     }
 
     /**
+     * Get Query Data Mock
+     *
+     * @return array
+     */
+    private function getQueryDataMock(): array
```
Comment (ihor-sviziev) [inline app/code/Magento/Analytics/Test/Unit/ReportXml/QueryFactoryTest.php]: Please move all new private methods below public

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

**Review by ihor-sviziev — APPROVED** [review]

**Comment by ihor-sviziev** [comment]
Functional test failures not related to this PR

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-6894](https://jira.corp.magento.com/browse/ENGCOM-6894) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Pull request merged**